src/element-transformer/ElementTransformer.ts

```diff
@@ -206,100 +235,65 @@ export default function elementTransformer<T extends ts.Node>(
 				if (initializer && ts.isCallExpression(initializer)) {
 					const call = initializer as ts.CallExpression;
 					const renderOptionsCallback = call.arguments[0];
-					let typeOfOptions: ts.Type | undefined;
-					if (
-						renderOptionsCallback &&
-						ts.isFunctionLike(renderOptionsCallback) &&
-						renderOptionsCallback.parameters[0]
-					) {
-						typeOfOptions = checker.getTypeAtLocation(renderOptionsCallback.parameters[0]);
-					} else if (renderOptionsCallback && ts.isIdentifier(renderOptionsCallback)) {
-						const functionType = checker.getTypeAtLocation(renderOptionsCallback);
-						const signatures = functionType.getCallSignatures();
-						for (const signature of signatures) {
-							if (signature.getParameters()[0]) {
-								typeOfOptions = checker.getTypeOfSymbolAtLocation(
-									signature.getParameters()[0],
-									renderOptionsCallback
-								);
-								if (typeOfOptions.getProperty('properties')) {
-									break;
-								}
-							}
-						}
-					}
 
-					if (typeOfOptions && typeOfOptions.getProperty('properties')) {
-						const optionProperties = typeOfOptions.getProperty('properties');
-						const typeOfProperties =
-							optionProperties && checker.getTypeOfSymbolAtLocation(optionProperties, node);
-
-						if (typeOfProperties && typeOfProperties.getCallSignatures()) {
-							const propertyCallSignatures = typeOfProperties.getCallSignatures();
-							let propertyType;
-							for (const propertyCallSignature of propertyCallSignatures) {
-								if (!propertyCallSignature.getParameters().length) {
-									propertyType = propertyCallSignature.getReturnType();
-								}
+					const propertyType = getPropertiesOfFactory(call);
```

Comment:
This is not new code, I just deleted some surrounding conditionals since they moved up to the helper function.